### PR TITLE
Add classref documentation for AnimationLibrary

### PR DESCRIPTION
**Pull request by jtnicholl**
Add class, method, and signal descriptions for AnimationLibrary, since it has none, as well as a tutorial link for the animation tutorial index (even though it doesn't actually cover animation libraries yet).
Writing this I noticed there's an entry for `_data` in the XML file. It doesn't actually show up in the docs, and running `--doctool` adds it back if it's removed. I don't know it that's an issue or if it's supposed to be there.

### CODE REVIEW

**Comment by jtnicholl** [comment]
Just rebased and updated to make use of the new `param` tags.

File: doc/classes/AnimationLibrary.xml [hunk 1]
```diff
@@ -1,47 +1,56 @@
...
 			<param index="0" name="name" type="StringName" />
 			<description>
+				Returns [code]true[/code] if the library stores an [Animation] with [param name] as the key.
 			</description>
 		</method>
 		<method name="remove_animation">
 			<return type="void" />
 			<param index="0" name="name" type="StringName" />
 			<description>
+				Removes the [Animation] with they key [param name].
```
Comment (YuriSizov) [inline doc/classes/AnimationLibrary.xml:45]:
```suggestion
				Removes the [Animation] with the key [param name].
```

File: doc/classes/AnimationLibrary.xml [hunk 2]
```diff
@@ -1,47 +1,56 @@
...
 			</description>
 		</method>
 		<method name="rename_animation">
 			<return type="void" />
 			<param index="0" name="name" type="StringName" />
 			<param index="1" name="newname" type="StringName" />
 			<description>
+				Changes the [Animation] associated with the key [param name] to be stored instead under the key [param newname].
```
Comment (YuriSizov) [inline doc/classes/AnimationLibrary.xml:53]: This is a bit wordy and worded in a confusing way.

```suggestion
				Changes the key of an [Animation] associated with the key [param name] to [param newname].
```

**Comment by YuriSizov** [comment]
Documentation seems fine, besides the two notes I've left. Would be nice if @godotengine/animation confirmed that the descriptions are factually correct.

**Comment by TokageItLab** [comment]
Seems generally good. As for nitpick, it can be added information about the global animation library.

**Review by mhilbrunner — APPROVED** [review]

**Comment by mhilbrunner** [comment]
> Are you talking about the library with an empty StringName as the key? Doesn't that belong under AnimationPlayer's docs, not AnimationLibrary's?

I'm not entirely clear on this, but seeing how long this one has been sitting here, and seeing that it looks good, I'm going to go ahead and merge it; further improvements can always be made iteratively in more PRs. Thanks for working on docs!
